autosynth/synth.py

```diff
@@ -519,82 +519,93 @@ def _inner_main(temp_dir: str) -> int:
+        multiple_prs = flags[autosynth.flags.AUTOSYNTH_MULTIPLE_PRS]
+        if (not multiple_commits and not multiple_prs) or not metadata:
+            if change_pusher.check_if_pr_already_exists(branch):
+                return 0
+
+            synth_log = Synthesizer(
+                metadata_path,
+                args.extra_args,
+                deprecated_execution=args.deprecated_execution,
+            ).synthesize(base_synth_log_path)
+
+            if not has_changes():
+                logger.info("No changes. :)")
+                sys.exit(EXIT_CODE_SKIPPED)
 
-        # Call the loop.
-        commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
+            git.commit_all_changes(pr_title)
+            change_pusher.push_changes(1, branch, pr_title, synth_log)
+            return 1
 
-        if commit_count == 0:
-            logger.info("No changes. :)")
-            sys.exit(EXIT_CODE_SKIPPED)
+        else:
+            if not multiple_prs and change_pusher.check_if_pr_already_exists(branch):
+                return 0  # There's already an existing PR
+
+            # Enumerate the versions to loop over.
+            sources = metadata.get("sources", [])
+            source_versions = [
+                git_source.enumerate_versions_for_working_repo(metadata_path, sources)
+            ]
+            # Add supported source version types below:
+            source_versions.extend(
+                git_source.enumerate_versions(sources, pathlib.Path(temp_dir))
+            )
+
+            # Prepare to call synthesize loop.
+            synthesizer = Synthesizer(
+                metadata_path, args.extra_args, args.deprecated_execution, "synth.py",
+            )
+            x = SynthesizeLoopToolbox(
+                source_versions,
+                branch,
+                temp_dir,
+                metadata_path,
+                args.synth_path,
+                base_synth_log_path,
+            )
+            if not multiple_commits:
+                change_pusher = SquashingChangePusher(change_pusher)
 
-        return commit_count
+            # Call the loop.
+            commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
+
+            if commit_count == 0:
+                logger.info("No changes. :)")
+                sys.exit(EXIT_CODE_SKIPPED)
+
+            return commit_count
+    finally:
+        if args.synth_path:
+            # We're generating code in a mono repo.  The state left behind will
+            # probably be useful for generating the next API.
+            pass
+        else:
+            # We're generating a single API in a single repo, and using a different
+            # repo to generate the next API.  So the next synth will not be able to
+            # use any of this state.  Clean it up to avoid running out of disk space.
+            executor.check_call(["git", "clean", "-fdx"], cwd=working_repo_path)
```

Comment:
@chingor13 Will this wipe out the log files?  Where do they live?

Reply:
The logs live in autosynth's `logs/` directory at the root (sibling of the working repo).